unittests/savanna_misc_tests.cpp

```diff
@@ -585,12 +584,107 @@ BOOST_FIXTURE_TEST_CASE(validate_qc_requiring_finalizer_policies, savanna_cluste
+   auto p2 = pending->generation;                       // and its generation is higher than the active one
+   BOOST_REQUIRE_EQUAL(p2, p1 + 1);                     // b3 has new pending finalizer policy p2
+
+   const std::vector<size_t> partition {0};             // partition A so that B, C and D don't see b4 (yet)
+   set_partition(partition);                            // and don't vote on it
+
+   // push action so that the block is not empty
+   A.push_action(config::system_account_name, updateauth::get_name(), tester_account,
+                 fc::mutable_variant_object()("account", "tester")("permission", "first")("parent", "active")(
+                    "auth", authority(A.get_public_key(tester_account, "first"))));
+
+   auto b4 = A.produce_block();
+   print("b4", b4);
+   BOOST_REQUIRE_EQUAL(qc_s(qc(b4)), strong_qc(b3));    // b4 claims a strong QC on b3
+   BOOST_REQUIRE_EQUAL(A.lib_number, b2->block_num());  // b4 makes B2 final
+
+   auto b5 = A.produce_block();
+   print("b5", b5);
+   BOOST_REQUIRE(!qc(b5));                              // b5 doesn't include a new qc (duplicates b4's strong claim on b3)
+   BOOST_REQUIRE_EQUAL(A.lib_number, b2->block_num());  // finality unchanged stays at b2
+
+   set_partition({});                                   // remove partition so A will receive votes on b4 and b5
+
+   push_block(0, b4);                                   // other nodes receive b4 and vote on it, so A forms a qc on b4
+   auto b6 = A.produce_block();
+   print("b6", b6);
+   BOOST_REQUIRE_EQUAL(qc_s(qc(b6)), strong_qc(b4));    // b6 claims a strong QC on b4
+   BOOST_REQUIRE_EQUAL(A.lib_number, b3->block_num());  // b6 makes b3 final.
+
+   push_block(0, b5);
+
+   auto b7 = A.produce_block();
+   print("b7", b7);
+   BOOST_REQUIRE_EQUAL(qc_s(qc(b7)), strong_qc(b5));    // b7 claims a strong QC on b5
+   BOOST_REQUIRE_EQUAL(A.lib_number, b3->block_num());  // lib is still b3
+
+   push_block(0, b6);                                   // push b6 again as it was unlinkable until the other
+                                                        // nodes received b5
+
+   auto b8 = A.produce_block();
+   print("b8", b8);
+   BOOST_REQUIRE_EQUAL(qc_s(qc(b8)), strong_qc(b6));    // b8 claims a strong QC on b6
+   BOOST_REQUIRE_EQUAL(A.lib_number, b4->block_num());  // b8 makes B4 final
+
+   push_block(0, b7);                                   // push b7 and b8 as they were unlinkable until the other
+   push_block(0, b8);                                   // nodes received b6
+
+   auto b9 = A.produce_block();
+   print("b9", b9);
+   BOOST_REQUIRE_EQUAL(qc_s(qc(b9)), strong_qc(b8));    // b9 claims a strong QC on b8
+   BOOST_REQUIRE_EQUAL(A.lib_number, b6->block_num());  // b9 makes B6 final
+
+   // check that the block id of b9 match what we got with spring 1.0
+   auto b9_id = b9->calculate_id();
+   BOOST_REQUIRE_EQUAL(b9_id, block_id_type{"00000013725f3d79bd4dd4091d0853d010a320f95240981711a942673ad87918"});
```

Comment:
This is okay for now. But not quite what was requested in https://github.com/AntelopeIO/spring/issues/694.

The goal for this test was to commit the blocks generated by Spring 1.0.0 to the repo and then have the test successfully synchronize the blocks. That way if there was a problem, the error would help us reveal where the problem was. And we could always step-through debug to get the bottom of the issue.

With this approach we just detect something is wrong but the test does not permit us to figure out what could be the cause.

Anyway, you do not need to do what was originally requested as part of this issue. This achieves the goal of detecting unintentional consensus changes. I created a new issue https://github.com/AntelopeIO/spring/issues/742 to track the further work needed to augment this test, and that work can go to `main` instead.